src/lib/coda_graphql/coda_graphql.ml

```diff
@@ -667,6 +667,46 @@ module Types = struct
             ~resolve:(fun _ {coinbase; _} -> Currency.Amount.to_uint64 coinbase)
         ] )
 
+  let protocol_state_proof : (Coda_lib.t, Proof.t option) typ =
+    let display_g1_elem (g1 : Crypto_params.Tick_backend.Inner_curve.t) =
+      let x, y = Crypto_params.Tick_backend.Inner_curve.to_affine_exn g1 in
+      List.map [x; y] ~f:Crypto_params.Tick0.Field.to_string
+    in
+    let display_g2_elem (g2 : Curve_choice.Tock_full.G2.t) =
+      let open Curve_choice.Tock_full in
+      let x, y = G2.to_affine_exn g2 in
+      let to_string (fqe : Fqe.t) =
+        let vector = Fqe.to_vector fqe in
+        List.init (Fq.Vector.length vector) ~f:(fun i ->
+            (* TODO: visually display string better *)
+            let string_value =
+              Binable.to_string (module Fq) (Fq.Vector.get vector i)
```

Comment:
`Crypto_params.Tick0.Field.to_string (Fq.Vector.get vector i)` might be good to use here as well.

Reply:
okay sweet, this works. Thanks matt :)